### PR TITLE
analytics page + autoformatted javascript files 

### PR DESCRIPTION
**Pull request by powahftw**

### CODE REVIEW

File: static/js/settings.js [hunk 1]
```diff
@@ -124,45 +117,48 @@ const getAndRenderSettingsPage = async () => {
...
                  Disk free space: ${responseData.free_space}/${responseData.total_space} GiB
                  </p>`;
 };
 
 const updateSettings = async () => {
   const settings = {};
-  idToJsonFields.forEach((field) => {
-    jsonField = field;
+  idToJsonFields.forEach((jsonField) => {
     idField = jsonToIdFields.get(field);
```
Comment (mvkv) [inline static/js/settings.js]: change "field" to "jsonField" also here

File: templates/analytics.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{% set active_page = 'analytics' %}
+{% extends 'base.html' %} 
+
+{% block content %}
+<div class="container analytics">  
+    <p class="breadcrumb"><a href="/analytics">Analytics</a></p>
```
Comment (mvkv) [inline templates/analytics.html:6]: Give the link a white color and remove text decorations

File: flask_server.py [hunk 3]
```diff
@@ -182,6 +204,45 @@ def delete_media():
...
+    return jsonify({"selected_user": None, "all_users": users_with_json_files, "json_file": None})
+
+@app.route("/api/analytics/<user_id>/", methods=['GET'])
+def analytics_api_single_user(user_id):
+    error_response = {"selected_user": None, "all_users": [], "user_json_file": None}
+    media_path = settings.get("media_folder_path")
+    json_file_path = os.path.join(os.path.join(media_path, user_id), f'{user_id}.json')
+    # Check we have metadata of this user. Not having this might mean we didn't saved the .json stories or the folder structure is an old one.
+    if not (os.path.exists(json_file_path)):
+        logger.info(f"Didn't {json_file_path} while looking for analytics files")
```
Comment (mvkv) [inline flask_server.py]: Add "found"

File: static/js/analytics.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+};
+
+const renderChartsFromSingleUserJson = (json) => {
+  renderMediaCountGraphFromJson(json);
+  renderHourlyFrequencyGraphFromJson(json);
+  renderTaggedFriendsGraphFromJson(json);
+};
+
+const getAndRenderAnalytics = async () => {
+  const baseUrl = window.location.origin;
```
Comment (mvkv) [inline static/js/analytics.js]: pick this one from the top of the file

File: static/js/analytics.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+  pageHtml += "</div>";
+  return `<div class='analitics-wrapper'>${pageHtml}</div>`;
+};
+
+const taggedFriendsFromJson = (json) => {
+  return json
+    .filter((j) => "reel_mentions" in j)
+    .flatMap((j) => j["reel_mentions"])
+    .filter((j) => j["display_type"] === "mention_username")
+    .map((j) => j["user"]["full_name"] ?? j["user"]["username"])
```
Comment (mvkv) [inline static/js/analytics.js:43]: Show also the username in the graph

**Review by mvkv — APPROVED** [review]

File: static/js/analytics.js (hunk 4 above)
Comment (powahftw) [reply static/js/analytics.js]: Done!

File: flask_server.py (hunk 3 above)
Comment (powahftw) [reply flask_server.py]: Done

File: static/js/analytics.js (hunk 5 above)
Comment (powahftw) [reply static/js/analytics.js:43]: Avoided due to space issues. Might consider going only 'username' in the future

File: static/js/settings.js (hunk 1 above)
Comment (powahftw) [reply static/js/settings.js]: ?

File: templates/analytics.html (hunk 2 above)
Comment (powahftw) [reply templates/analytics.html:6]: I went with this option as it make it clearer it's clickable.

**Pull request merged**